oam/install_scripts/columnstore-pre-uninstall.in

```diff
@@ -13,41 +13,15 @@ running_systemd() {
 
 }
 
-rpmmode=install
-user=`whoami 2>/dev/null`
-
-quiet=0
-
-for arg in "$@"; do
-	if [ `expr -- "$arg" : '--rpmmode='` -eq 10 ]; then
-		rpmmode="`echo $arg | awk -F= '{print $2}'`"
-	elif [ `expr -- "$arg" : '--user='` -eq 7 ]; then
-		user="`echo $arg | awk -F= '{print $2}'`"
-	elif [ `expr -- "$arg" : '--quiet'` -eq 7 ]; then
-		quiet=1
-	else
-		echo "pre-install: invalid unknown argument: $arg" 1>&2
-		echo "exiting..."
-		exit 1
-	fi
-done
-
-#stop services
-columnstore stop > /dev/null 2>&1
-
 # Test we are using systemd
-systemctl cat mariadb.service > /dev/null 2>&1
+systemctl cat mariadb-columnstore.service > /dev/null 2>&1
 if [ $? -eq 0 ] && [ $(running_systemd) -eq 0 ]; then
-    systemctl stop mariadb.service > /dev/null 2>&1
+    systemctl stop mariadb-columnstore >/dev/null 2>&1
 else
-    pkill mysqld
-fi
-
-cloud=`mcsGetConfig Installation Cloud`
-if [ $cloud = "amazon-ec2" ] || [ $cloud = "amazon-vpc" ]; then
-	if test -f /etc/fstab ; then
-	    sed -i '/Columnstore\/data/d' /etc/fstab > /dev/null 2>&1
-	fi
+    PROGS='load_brm workernode controllernode PrimProc ExeMgr DMLProc DDLProc WriteEngineServer'
```

Comment:
The short answer we need to have an interface in SM, WriteEngine, DMLProc, DDLProc to answer if there are any write operations running.

Reply:
W/o it any delay isn't good enough as the recent cloud issue showed

Reply:
If SM is running after sending sigterm, it's still writing.  It goes away when it's done.

Reply:
The others, I suspect are the same but I can't say for sure.

Reply:
if/when we add SM, it could take a long time to flush, we'll need a death-detector to optimize this.  Actually, the reason for the current problem in sky is SM is being killed with -9, and it can't recover from that right now.